### PR TITLE
Ctrl for sequence classification

### LINKED ISSUE #7623: Implement PyTorch and/or TensorFlow sequence classification architectures for causal language models
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow (cf. issues #7622), but it would also be great to have it for other causal models: ~OpenAI GPT~, ~CTRL~ (PR opened @elk-cloner), ~TransfoXL~ (PR opened @spatil6)

Below is a list of items to follow to make sure the integration of such an architecture is complete:

- Implement `XXXForSequenceClassification` in `modeling_xxx.py` or `TFXXXForSequenceClassification` in `modeling_tf_xxx.py
- Test that architecture in `tests/test_modeling_xxx.py` or `tests/test_modeling_tf_xxx.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/xxx.rst`.

Taking a look at the code changes in #7501 would be a good start.

A very good first issue to get acquainted with the library and its architectures!

### PR DESCRIPTION
**Pull request by elk-cloner**
# What does this PR do?

Fixes #7623

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/ctrl/modeling_ctrl.py [hunk 1]
```diff
@@ -571,3 +571,118 @@ def forward(
...
+
+        hidden_states = transformer_outputs[0]
+        logits = self.classifier(hidden_states)
+
+        if input_ids is not None:
+            batch_size, sequence_length = input_ids.shape[:2]
+        else:
+            batch_size, sequence_length = inputs_embeds.shape[:2]
+
+        # TODO: it is not working right now, i think we should set config.pad_token_id=1 not None
```
Comment (elk-cloner) [inline src/transformers/models/ctrl/modeling_ctrl.py:650]: i'm not sure about this one. @LysandreJik can you help me ?

Comment (LysandreJik) [reply src/transformers/models/ctrl/modeling_ctrl.py:650]: When you say it isn't working right now, you mean that this assertion raises when the batch size is superior to 1 and no padding token is defined?

Comment (elk-cloner) [reply src/transformers/models/ctrl/modeling_ctrl.py:650]: yes exactly. for example if you run `test_attention_outputs` bache size is superior to 1 and padding token is not define. i checked it and found out that default value for padding token in config is `None`

Comment (LysandreJik) [reply src/transformers/models/ctrl/modeling_ctrl.py:650]: The configuration object should remain as it is, but you can update the `CTRLModelTester` to define the `pad_token_id` as it was done for the GPT-2 model: https://github.com/huggingface/transformers/blob/master/tests/test_modeling_gpt2.py#L93

Comment (elk-cloner) [reply src/transformers/models/ctrl/modeling_ctrl.py]: i fixed it.

**Review by LysandreJik — APPROVED** [review]
LGTM, thank you!

**Comment by LysandreJik** [comment]
Same as GPT-2, this would benefit from also handling padding on the left; I'll work on this in another PR.

**Review by sgugger — APPROVED** [review]
Thanks for adding this new model!

**Review by patrickvonplaten — APPROVED** [review]
Very clean! Thanks for adding the model
